templates/coderd.yaml

```diff
@@ -198,6 +198,34 @@ data:
   privateKey: {{ genPrivateKey "ecdsa" | b64enc }}
   {{- end }}
 {{- end }}
+{{ if eq (include "coder.useServiceNext" .) "true" }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  {{- if (lookup "v1" "Service" .Release.Namespace "ingress-nginx") }}
+  name: nginx-ingress
+  {{- else }}
+  name: {{ include "coder.serviceName" . }}
+  {{- end }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+{{ toYaml .Values.coderd.serviceSpec | indent 2 }}
+  selector:
+    coder.deployment: {{ include "coder.serviceName" . }}
+  ports:
+    - name: tcp-{{ include "coder.serviceName" . }}
+      port: 80
+      targetPort: 8080
+      protocol: TCP
```

Comment:
while you're in here, do you also want to add an `appProtocol`? https://kubernetes.io/docs/concepts/services-networking/service/#application-protocol

This was introduced as beta in 1.19, so I think should be OK given our version support policy https://v1-19.docs.kubernetes.io/docs/concepts/services-networking/service/#application-protocol